helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/BestPossibleExternalViewVerifier.java

```diff
@@ -377,8 +385,12 @@ private BestPossibleStateOutput calcBestPossState(ResourceControllerDataProvider
     }
 
     runStage(event, new CurrentStateComputationStage());
+    DryrunWagedRebalancer wagedRebalancer = new DryrunWagedRebalancer(_zkClient.getServers(), cache.getClusterName(),
+        cache.getClusterConfig().getGlobalRebalancePreference());
     // TODO: be caution here, should be handled statelessly.
-    runStage(event, new BestPossibleStateCalcStage());
+    runStage(event, new BestPossibleStateCalcStage(wagedRebalancer));
+
+    wagedRebalancer.close();
```

Comment:
That would be OK, I think. If any exception, we should end the verifier otherwise it might consume all the resources in the loop.

AutoCloseable only works if we use try-with-resource, right? How it can help if we are not using a try-with-resource block?

Reply:
Synced with @pkuwm offline. Now I got what he meant.